ResourceStatusSystem/Utilities/RSSCacheNoThread.py

```diff
@@ -321,25 +326,38 @@ def __match( self, validCache, elementNames, elementType, statusTypes ):
     if isinstance( elementNames, str ):
       elementNames = [ elementNames ]
     elif elementNames is None:
-      elementNames = [ cacheKey[0] for cacheKey in cacheKeys ]
+      if isinstance(cacheKeys[0], (tuple, list)):
+        elementNames = [ cacheKey[0] for cacheKey in cacheKeys ]
+      else:
+        elementNames = cacheKeys
     # Remove duplicates, makes Cartesian product faster
     elementNamesSet = set( elementNames )
 
     if isinstance( elementType, str ):
-      elementType = [ elementType ]
+      if not elementType or elementType == 'Site':
+        elementType = ''
+      else:
+        elementType = [ elementType ]
     elif elementType is None:
       elementType = [ cacheKey[1] for cacheKey in cacheKeys ]
     # Remove duplicates, makes Cartesian product faster
     elementTypeSet = set( elementType )
 
     if isinstance( statusTypes, str ):
-      statusTypes = [ statusTypes ]
+      if not statusTypes:
```

Comment:
`statusTypes = [statusTypes] if statusTypes else ''`
but this sounds strange as the type is different... Shouldn't it be `[]`

Reply:
correct. I will change it.